cmds/networkd/addr_watcher.go

```diff
@@ -56,6 +57,43 @@ func (w WatchedLinks) CallBack(update netlink.AddrUpdate) error {
 	return publishIfaces(ifaces, w.nodeID, w.bcdb)
 }
 
+func (w WatchedLinks) Start(ctx context.Context) error {
+	ch := make(chan netlink.AddrUpdate)
+	done := make(chan struct{})
+	defer close(done)
+	cErr := make(chan error)
+
+	log.Info().Msg("start netlink addr subscription")
+	if err := netlink.AddrSubscribe(ch, done); err != nil {
+		return err
+	}
+
+	go func() {
+		for {
+			select {
+
+			case <-ctx.Done():
+				cErr <- nil
+
+			case update, ok := <-ch:
+				if !ok {
+					log.Error().Msg("netlink address subscription exited")
+					cErr <- fmt.Errorf("netlink closed the subscription channel")
+					return
+				}
+
+				log.Debug().Msgf("addr update received %+v", update)
+				if err := w.callBack(update); err != nil {
+					log.Error().Err(err).Msg("addr watcher: error during callback")
+				}
+			}
+		}
+	}()
+
+	err := <-cErr
```

Comment:
If u waiting on an error anyway (not returing) why this has to be in a go routine ?